components/cdc/src/service.rs

```diff
@@ -346,7 +348,11 @@ impl ChangeData for Service {
 
         let peer = ctx.peer();
         let scheduler = self.scheduler.clone();
+        let cluster_id = self.cluster_id;
         let recv_req = stream.try_for_each(move |request| {
+            if cluster_id != request.get_header().get_cluster_id() {
```

Comment:
It's not necessary to check it for every item in the stream. How about using `RpcContext::request_headers`?

Reply:
I prefer checking cluster id in request's header, even though it needs to be checked for every stream. 
1. checking cluster id is not performance bottleneck.
2. it's more consistent with other checking, no magic header keys.